tests/test_tags.py

```diff
@@ -179,29 +175,226 @@ def test_tagify_deep_copy():
+            """\
+            <div>
+              a<span>b</span>
+            </div>"""
+        ),
+    )
+    expect_html(
+        div(span("a"), span("b")),
+        textwrap.dedent(
+            """\
+            <div>
+              <span>a</span><span>b</span>
+            </div>"""
+        ),
+    )
+
+    # Block tags with one block and one inline child
+    expect_html(
+        div("a", div("b")),
+        textwrap.dedent(
+            """\
+            <div>
+              a
+              <div>b</div>
+            </div>"""
+        ),
+    )
+    expect_html(
+        div(span("a"), div("b")),
+        textwrap.dedent(
+            """\
+            <div>
+              <span>a</span>
+              <div>b</div>
+            </div>"""
+        ),
+    )
+    expect_html(
+        div(div("a"), "b"),
+        textwrap.dedent(
+            """\
+            <div>
+              <div>a</div>
+              b
+            </div>"""
+        ),
+    )
+    expect_html(
+        div(div("a"), span("b")),
+        textwrap.dedent(
+            """\
+            <div>
+              <div>a</div>
+              <span>b</span>
+            </div>"""
+        ),
+    )
+
+    # Block tag with two block children
+    expect_html(
+        div(div("a"), div("b")),
+        textwrap.dedent(
+            """\
+            <div>
+              <div>a</div>
+              <div>b</div>
+            </div>"""
+        ),
+    )
+
+    # Block tag with three children; mix of inline and block
+    expect_html(
+        div(span("a"), span("b"), div("c")),
+        textwrap.dedent(
+            """\
+            <div>
+              <span>a</span><span>b</span>
+              <div>c</div>
+            </div>"""
+        ),
+    )
+    expect_html(
+        div(span("a"), "b", div("c")),
+        textwrap.dedent(
+            """\
+            <div>
+              <span>a</span>b
+              <div>c</div>
+            </div>"""
+        ),
+    )
+    expect_html(
+        div(div("a"), "b", span("c")),
+        textwrap.dedent(
+            """\
+            <div>
+              <div>a</div>
+              b<span>c</span>
+            </div>"""
+        ),
+    )
+
+    # More complex nesting
+    expect_html(
+        div(span(tags.b("a")), span(tags.b("b"))),
+        textwrap.dedent(
+            """\
+            <div>
+              <span><b>a</b></span><span><b>b</b></span>
+            </div>"""
+        ),
+    )
+    expect_html(
+        span(span(tags.b("a")), span(tags.b("b")), span("c")),
+        textwrap.dedent(
+            """\
+            <span><span><b>a</b></span><span><b>b</b></span><span>c</span></span>"""
+        ),
+    )
+    expect_html(
+        div(div(span("a")), span(tags.b("b"))),
+        textwrap.dedent(
+            """\
+            <div>
+              <div>
+                <span>a</span>
+              </div>
+              <span><b>b</b></span>
+            </div>"""
+        ),
+    )
+
```

Comment:
Maybe it'd be worth adding some test cases for `TagList()` (e.g., `TagList("a", "b", div("c", "d"), span("e", "f"), span("g", "h"))`)?